File: tests/fixtures/generic/vulnerable/insecure_cookie.cs
// @audit-fixture
// @rule: insecure_cookie
// @kind: vulnerable
// VULNERABLE: Response.Cookies.Append sans CookieOptions{ Secure=true, HttpOnly=true }
using Microsoft.AspNetCore.Mvc;

public class LoginController : Controller {
    public IActionResult Login() {
        Response.Cookies.Append("session", "abc123");
        return Ok();
    }
}

File: tests/fixtures/generic/vulnerable/insecure_cookie.js
// @audit-fixture
// @rule: insecure_cookie
// @kind: vulnerable
// VULNERABLE: cookie sans httpOnly/secure/sameSite — accessible en JS (XSS)
//             et envoyé en HTTP non chiffré (CWE-614).
const express = require("express");
const app = express();
app.post("/login", (req, res) => {
    res.cookie("session", "abc123");
    res.send("OK");
});
